addons/website_sale/controllers/main.py

```diff
@@ -200,6 +200,12 @@ def shop(self, page=0, category=None, search='', **post):
 
         product_obj = pool.get('product.template')
 
+        def recursive_category(category, parent_category=[]):
```

Comment:
Thanks!

For your info, I have backported this functionality to v8 here: https://github.com/OCA/website/pull/20